test/metabase/lib/test_util/generators.cljc

```diff
@@ -74,6 +76,22 @@
 (defn- add-step [{:keys [kind] :as step-def}]
   (swap! step-kinds assoc kind step-def))
 
+;; Helpers =======================================================================================
+(defn- choose
+  "Uniformly chooses among a seq of options.
+
+  Returns nil if the list is empty! This is handy for choose-and-do vs. do-nothing while writing the next steps."
+  [xs]
+  (when-not (empty? xs)
+    (rand-nth xs)))
+
+(defn- choose-stage
+  "Chooses a stage to operator on. 80% act on -1, 20% chooses a stage by index (which might be the last stage)."
```

Comment:
```suggestion
  "Chooses a stage to operate on. 80% act on -1, 20% chooses a stage by index (which might be the last stage)."
```

Reply:
That one actually got fixed in the later filters branch, so I'll just leave it for that PR.